frame_transformer/include/frame_transformer/frame_transformer.hpp

```diff
@@ -0,0 +1,138 @@
+    Transformer(Config config, std::shared_ptr<tf2_ros::Buffer> buffer, std::shared_ptr<carma_ros2_utils::CarmaLifecycleNode> node)
+    : TransformerBase(config, buffer, node) {
+
+      input_sub_ = node_->create_subscription<T>("input", config_.queue_size,
+                                                  std::bind(&Transformer::input_callback, this, std_ph::_1));
+
+      // Setup publishers
+      output_pub_ = node_->create_publisher<T>("output", config_.queue_size);
+    }
+
+
+    /**
+     * \brief Helper method which takes in an input message and transforms it to the provided frame.
+     *        Returns false if the provided timeout is exceeded for getting the transform or the transform could not be computed
+     * 
+     * \param in The input message to transform
+     * \param[out] out The preallocated location for the output message
+     * \param target_frame The frame the out message data will be in
+     * \param timeout A timeout in ms which if exceeded will result in a false return and invalid out value. This call may block for this period.
+     *                If set to zero, then lookup will be attempted only once. 
+     * 
+     * \return True if transform succeeded, false if timeout exceeded or transform could not be performed. 
+     */ 
+    bool transform(const T &in, T &out, const std::string &target_frame, const std_ms timeout)
+    {
+
+      try
+      {
+        buffer_->transform(in, out, target_frame, timeout);
+      }
+      catch (tf2::TransformException &ex)
+      {
+        std::string error = ex.what();
+        error = "Failed to get transform with exception: " + error;
+        auto& clk = *node_->get_clock(); // Separate reference required for proper throttle macro call
+        RCLCPP_WARN_THROTTLE(node_->get_logger(), clk, 1000, error);
+        
+        return false;
+      }
+
+      return true;
+    }
+
+    /**
+     * \brief Callback for input data. Transforms the data then republishes it
+     * 
+     * NOTE: This method can be specialized for unique preallocation approaches for large messages such as point clouds or images
+     * 
+     * \param in_msg The input message to transform
+     */ 
+    void input_callback(std::unique_ptr<T> in_msg)
+    {
+      T out_msg;
+
+      if (!transform(*in_msg, out_msg, config_.target_frame, std_ms(config_.timeout)))
+      {
+        return;
+      }
+
+      output_pub_->publish(out_msg);
+    }
+  };
+
+  // Specialization of input_callback for PointCloud2 messages to allow for preallocation of points vector
+  // This is done due to the large size of that data set
+  template <>
+  inline void Transformer<sensor_msgs::msg::PointCloud2>::input_callback(std::unique_ptr<sensor_msgs::msg::PointCloud2> in_msg) {
+
+    sensor_msgs::msg::PointCloud2 out_msg;
+    out_msg.data.reserve(in_msg->data.size()); // Preallocate points vector
+
+    if (!transform(*in_msg, out_msg, config_.target_frame, std_ms(config_.timeout)))
+    {
+      return;
```

Comment:
Would it make sense to log an error here?

Reply:
The log already occurs in the transform() method. Logging here would be redundant.